src/main/java/org/aksw/simba/lemming/ColouredGraphDecorator.java

```diff
@@ -0,0 +1,300 @@
+/**
+ * 
+ */
+package org.aksw.simba.lemming;
+
+import org.aksw.simba.lemming.mimicgraph.constraints.TripleBaseSingleID;
+import org.aksw.simba.lemming.util.IntSetUtil;
+
+import com.carrotsearch.hppc.BitSet;
+
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import it.unimi.dsi.fastutil.ints.IntSet;
+
+/**
+ * Base Decorator for Coloured Graph Class
+ * 
+ * @author Pranav
+ */
+public class ColouredGraphDecorator implements IColouredGraph {
+
+    protected IColouredGraph graph;
+    protected boolean isAddingEdge;
+    protected TripleBaseSingleID triple;
+
+    public ColouredGraphDecorator() {
+        this.graph = null;
+        this.isAddingEdge = false;
+        this.triple = null;
+    }
+
+    /**
+     * Class constructor
+     * 
+     * @param isAddingEdge - represents the edge operation.Flag is true if edge is
+     *                     being added. Flag is false If edge is being removed.
+     */
+    public ColouredGraphDecorator(IColouredGraph graph, boolean isAddingEdgeFlag) {
+        this.graph = graph;
+        this.isAddingEdge = isAddingEdgeFlag;
+    }
+
+    public ColouredGraphDecorator(IColouredGraph graph) {
+        this.graph = graph;
+    }
+
+    /**
+     * Returns ColouredGraphDecorator object
+     * 
+     * @return ColouredGraphDecorator graph
+     */
+    public IColouredGraph getGraph() {
+        return this.graph;
+    }
+
+    /**
+     * Store current triple data
+     * 
+     * @param edge - Triple data
+     */
+    public void setTriple(TripleBaseSingleID edge) {
+        this.triple = edge;
+    }
+
+    /**
+     * Get current triple offered for adding or removing to the ColouredGraph
+     * 
+     * @return TripleBaseSingleID
+     */
+    public TripleBaseSingleID getTriple() {
+        return this.triple;
+    }
+
+    /**
+     * Get list of all Edge IDs connecting to vertex
+     *
+     * @param verticeId - verticeId the id of an vertex
+     * @return IntSet - set of edge IDs
+     */
+    @Override
+    public IntSet getEdgesIncidentTo(int verticeId) {
+        return this.graph.getEdgesIncidentTo(verticeId);
+    }
```

Comment:
Hint: When you override a method of a super class, an "own" javadoc is only necessary if you want to add an important implementation detail. The method will "inherit" the javadoc of the original method :wink: